janusgraph-core/src/main/java/org/janusgraph/graphdb/query/graph/GraphCentricQueryBuilder.java

```diff
@@ -81,13 +81,21 @@
      * Whether smart limit adjustment is enabled
      */
     private boolean useSmartLimit;
+    /**
+     * Selection service for the best combination of indexes to be queried
+     */
+    private IndexSelectionStrategy indexSelector;
 
     public GraphCentricQueryBuilder(StandardJanusGraphTx tx, IndexSerializer serializer) {
         Preconditions.checkNotNull(tx);
         Preconditions.checkNotNull(serializer);
         useSmartLimit = tx.getGraph().getConfiguration().adjustQueryLimit();
         this.tx = tx;
         this.serializer = serializer;
+        this.indexSelector =
+            new ThresholdBasedIndexSelectionStrategy(tx.getGraph().getConfiguration().getIndexSelectThreshold(),
+                new BruteForceIndexSelectionStrategy(),
+                new ApproximateIndexSelectionStrategy());
```

Comment:
I think we can add `indexSelector` as a dependency to this constructor instead of initializing it here to follow DI

Reply:
Are you ok with configuring it in `StandardJanusGraph`?

Reply:
Also, I guess `indexSelectThreshold` doesn't change from query to query, so we don't need to create a new implementation of a strategy each time. We should create indexSelectionStrategy only once when we establish a connection.

Reply:
I assume this is a "yes" :smile:

Reply:
Ah, I didn't see you past comment) Yes, configuring it in `StandardJanusGraph` looks good!